desc/geometry/core.py

```diff
@@ -441,14 +453,18 @@ def compute(
                 "must pass in a Grid object or an integer for argument grid!"
                 f" instead got type {type(grid)}"
             )
+
+        method = kwargs.pop("method", "auto")
         if params is None:
-            params = get_params(names, obj=self)
+            params = get_params(names, obj=self, basis=kwargs.get("basis", "rpz"))
         if transforms is None:
             transforms = get_transforms(
                 names,
                 obj=self,
                 grid=grid,
                 jitable=kwargs.pop("jitable", False),
```

Comment:
Does anyone know why we use `False` as the default here for the `Surface` class? For the `Curve` class we hard-code this to `True`.

Reply:
`jitable=True` means it uses the direct transforms, which for 1d curves is pretty cheap. For a 2d surface its a bit more expensive (especially if you need high resolution)